### PR TITLE
feat: proxy agent

### PR DESCRIPTION
**Pull request by divyenduz**
Adds HTTP proxy support

Fixes https://github.com/prisma/prisma2/issues/506

### CODE REVIEW

File: packages/fetch-engine/src/getProxyAgent.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+function formatHostname(hostname) {
+  // canonicalize the hostname, so that 'oogle.com' won't match 'google.com'
+  return hostname.replace(/^\.*/, '.').toLowerCase()
+}
+
+function parseNoProxyZone(zone) {
+  zone = zone.trim().toLowerCase()
+
+  var zoneParts = zone.split(':', 2)
```
Comment (timsuchanek) [inline packages/fetch-engine/src/getProxyAgent.ts]: Let's at least turn the `var` statements into `const` if possible or `let`

Comment (divyenduz) [reply packages/fetch-engine/src/getProxyAgent.ts]: Done, thanks

**Comment by janpio** [comment]
Does this justify docs changes? If so a follow up `kind/docs` issue should be created.

Also, is this reflected in any spec? It probably should be. Potential "CLI" spec, "Configuration" section maybe?

**Comment by divyenduz** [comment]
@janpio On my list to raise a PR for specs, create an issue for docs and this feature is not complete yet. Another PR will follow in Prisma 2 repo.

**Pull request merged**